src/coreclr/jit/fgflow.cpp

```diff
@@ -279,8 +279,14 @@ flowList* Compiler::fgRemoveRefPred(BasicBlock* block, BasicBlock* blockPred)
     noway_assert(block != nullptr);
     noway_assert(blockPred != nullptr);
 
-    noway_assert(block->countOfInEdges() > 0);
-    block->bbRefs--;
+    if (block->countOfInEdges() > 0)
```

Comment:
I don't understand under what conditions this function will be called and have bbRefs of zero. Maybe the caller in this case should do the check?

You're changing the invariants of this function, so you need to rewrite the header comment to match.

Reply:
Good point.

Likely there is some pred list updating being done in a caller that should be guarded by `fgComputePredsDone`.